core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -3,12 +3,18 @@ package syntax
 
 import cats.data.{Xor, XorT}
 
-trait ApplicativeErrorSyntax {
+trait ApplicativeErrorSyntax1 {
+  implicit def applicativeErrorSyntax[F[_], E, A](fa: F[A])(implicit F: ApplicativeError[F, E]): ApplicativeErrorOps[F, E, A] =
+    new ApplicativeErrorOps[F, E, A](fa)
+}
+
+trait ApplicativeErrorSyntax extends ApplicativeErrorSyntax1 {
   implicit def applicativeErrorIdSyntax[E](e: E): ApplicativeErrorIdOps[E] =
     new ApplicativeErrorIdOps(e)
 
-  implicit def applicativeErrorSyntax[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
+  implicit def applicativeErrorSyntax2[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
```

Comment:
Just checked with the SI-2712 fix, it works fine (can find syntax for `F[_, _]` with only one `F[_]` variant.  
Removing the existing `F[_, _]` variant, however, would mean we need to change existing syntax tests to use `Option` instead of `Xor`, right?